wlauto/common/linux/workload.py

```diff
@@ -0,0 +1,146 @@
+#    Copyright 2017 ARM Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+from math import ceil
+import time
+
+from wlauto.core.extension import Parameter
+from wlauto.core.workload import Workload
+from wlauto.core.resource import NO_ONE
+from wlauto.common.resources import File, Executable
+from wlauto.common.android.resources import ReventFile
+from wlauto.exceptions import WorkloadError, ResourceError, DeviceError
+from wlauto.utils.revent import ReventRecording
+
+class ReventWorkload(Workload):
+    # pylint: disable=attribute-defined-outside-init
+
+    parameters = [
+        Parameter('default_naming', kind=bool, default=False,
+                  description='Use the default naming. Set this option to true if the revent file you wish to use was '
+                              'generated using the wa record command.'),
+        Parameter('idle', kind=bool, default=False,
+                  description='Say whether or not you want this workload to be run in idle mode. In idle mode the revent '
+                              'will only run .setup and .close files and in between will be idle for however many seconds '
+                              'you specify.'),
+        Parameter('idle_time', kind=int, default=10,
+                  description='The time you wish the device to remain idle for (idle must be True for this to be relevant).'),
+        Parameter('file_name', kind=str, default=None,
+                  description='The name of the revent file to look for. Will also look for file_name.setup and file_name.close'),
+        Parameter('no_setup', kind=bool, default=False,
+                  description='Set to True to force disable the setup revent file from running.'),
+        Parameter('no_close', kind=bool, default=False,
```

Comment:
Also, I'm not sure these should be parameters. I would assume that the requirement for setup and teardown would depend on the workload, not the user; i.e. they should be configured on per-workload, rather than per-invocation, basis.

I would create class  attributes `setup_required` and `teardown_required` which default to `True` in the base and may be overridden by individual workload definitions as needed.